pandas/core/window/rolling.py

```diff
@@ -412,6 +415,19 @@ def _apply_tablewise(
         self._insert_on_column(out, obj)
         return out
 
+    def _center_window(self, result: np.ndarray, offset: int) -> np.ndarray:
+        """
+        Center the result in the window for weighted rolling aggregations.
+        """
+        if self.axis > result.ndim - 1:
+            raise ValueError("Requested axis is larger then no. of argument dimensions")
+
+        if offset > 0:
+            lead_indexer = [slice(None)] * result.ndim
+            lead_indexer[self.axis] = slice(offset, None)
+            result = np.copy(result[tuple(lead_indexer)])
+        return result
+
```

Comment:
In an effort to make the tests run again before introducing new ones, I encountered an error with "'Rolling' object has no attribute '_center_window'".
I tried to solve the error by adding the `_center_window()` method to the `BaseWindow` class. 
However, this resulted in new errors.

I am not sure whether it is correct to follow this debugging path. Could you quickly comment @jreback , @mroeschke  or @arw2019? 

Meanwhile, I am working on adding the changes you asked for. :)

Reply:
As mentioned in https://github.com/pandas-dev/pandas/pull/38780#discussion_r549834716, I think you shouldn't need any additional code in this `_apply` method.

The changes should just need to occur in `pandas/_libs/window/indexers.pyx` in the `calculate_variable_window_bounds` function to make it return indexes that "center" the window.

Reply:
For example, for non time offests, center is handled when calculating the indexes of each window

https://github.com/pandas-dev/pandas/blob/67d4cae17bec45e84b9cf51bcf4fb5bbe293b26f/pandas/core/window/indexers.py#L81

Reply:
> 
> 
> As mentioned in [#38780 (comment)](https://github.com/pandas-dev/pandas/pull/38780#discussion_r549834716), I think you shouldn't need any additional code in this `_apply` method.
> 
> The changes should just need to occur in `pandas/_libs/window/indexers.pyx` in the `calculate_variable_window_bounds` function to make it return indexes that "center" the window.

Just removed the unnecessary lines from rolling.py and that made the tests work again, thanks for that hint! At first I understood I had to migrate those lines to the `pandas/_libs/window/indexers.pyx` instead of ditching them all together.